### DISCUSSION TITLE
GitLab renovate-runner: Store image and services values in ENV variables

### DISCUSSION THREAD

**Discussion by jdfm** [renovatebot/renovate#12067] [Suggest an Idea]
Ahoy peeps! 

My team is currently setting up a scheduled pipeline based on the configs from https://gitlab.com/renovate-bot/renovate-runner/-/tree/main and it's been mostly smooth going so far. 

One of the issues we're running into right now is around pull rate limits from dockerhub, which can prevent the runner from working sometimes because it fails to pull the image it needs to run. The way to avoid this is to prepend any images that will be pulled from (usually by default) dockerhub with the contents of the `$CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX` ENV variable (https://docs.gitlab.com/ee/user/packages/dependency_proxy/). But, doing this in our `.gitlab-ci.yml` level would cause us to not inherit the latest configs from the official `renovate-runner` repo. 

Would it be possible to store the values currently being used by `image` and `services` into an ENV variable and then use that ENV variable when configuring the `image` and `services` tags in the yaml file so we can, on our end, prepend with the `$CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX` ENV variable? This would allow us to always be using the latest setup/configs from the official `renovate-runner` repo.

**Comment by rarkins** [comment]
As a non GitLab user I don't fully understand your use case/limitations, but could you check if https://docs.renovatebot.com/self-hosted-configuration/ helps in any way, if you haven't tried it already?

**Reply by jdfm** [reply]
This isn't relevant to `renovate` in a general sense, it's more about the `renovate-runner` found in the link I posted above, and specifically about the gitlab pipeline/job configs that exist there. I think I hadn't noticed that the issue tracker for renovate-runner was open and allows for non-maintainers and such to open a new one, I think I'll repost this discussion there.

**Reply by viceice** [reply]
Please stay here until we agreed a solution

**Reply by jdfm** [reply]
@viceice is what I wrote above enough context, or do you need some more information?

**Reply by viceice** [reply]
Yes, can you propose a draft MR on which we can work on?

Remember: we've two different renovate images, wich sometimes have a bigger gap between publishing.

**Reply by jdfm** [reply]
Just to clarify, the Draft MR would be on the gitlab `renovate-runner` repo, correct?

**Reply by viceice** [reply]
Yes

**Reply by jdfm** [reply]
https://gitlab.com/renovate-bot/renovate-runner/-/merge_requests/211 has the setup I was imagining and an example of what it would look like for peeps implementing `.gitlab-ci.yaml` on their end based on the templates defined in the original `renovate-runner` repo.

**Comment by jdfm** [comment]
The work for this has been merged into https://gitlab.com/renovate-bot/renovate-runner/

Thanks for the support!

**Reply by viceice** [reply]
Can you suggest a PR, so we use gitlab proxy for runner self-Update?

**Reply by jdfm** [reply]
For the purpose of the rate limiting issue, I think the base config is fine as it is right now. This should really only be relevant for those of us that have run into these rate limiting issues, anyone else should be fine with what comes by default now in the templates. We could add some documentation about rate limiting to the README and suggest the code pattern that I mentioned originally in the MR though, as a way of documenting why those values are stored in variables and how to use them.

The only other suggestion I could make for the base templates is to move the `image` and `services` configs into the `renovate` job itself so they're not top level. It would still allow anyone that wants to to have an organization level Renovate job that can run in a scheduled pipeline and such by forking the renovate-runner repo and adding their own custom configs, as we've done in our organization. But it would also allow peeps to, at the Target Repo level, have a dedicated renovate job by simply including the renovate templates at that level. As it is now, if we tried that, it would set the `image` and `services` settings to what's defined by the renovate templates, which might not be what we want.

**Reply by jdfm** [reply]
@viceice Maybe I misunderstood what you were requesting. Are you peeps running into rate limiting issues right now in the `renovate-runner` GitLab repo's pipelines? Or was the request to make it so the base config of `image` and `services` use the proxy by default? If it's the latter, then I think what I wrote above still stands. If it's the former, I can create that MR, it should be as straightforward as doing `image: $CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX/$CI_RENOVATE_IMAGE` and 
```
services: 
  - $CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX/$CI_RENOVATE_SERVICE
```

wherever it's relevant to do so in your `.gitlab-ci.yml` file in the `renovate-runner` repo.
